hw/ip/i2c/doc/i2c.md

```diff
@@ -0,0 +1,336 @@
+{{% lowrisc-doc-hdr I2C HWIP Technical Specification }}
```

Comment:
@martin-lueker 
sorry for the moving target.. but due to the hug migration, a lot of this has changed yet again.

Can you reference the markdown style guide? For example `toc` is no longer required, and we are now switching to standard `##` usage.